### PR TITLE
suppress IT_PRIVATE from response to DEDICATED diff

### PR DESCRIPTION
**Pull request by harshithpatte-g**
In the interconnect resource, for the interconnect type DEDICATED, the api is responding with IT_PRIVATE type.
The request is to consider IT_PRIVATE and DEDICATED as synonyms in terraform.
Thus this PR to suppress the diff when API responds with IT_PRIVATE for DEDICATED type.

More context is in b/65386683

**Release Note Template for Downstream PRs (will be copied)**

```release-note:enhancement
compute: fixed perma-diff for `interconnect_type` being `DEDICATED` in `google_compute_interconnect` resource
```

### CODE REVIEW

**Comment by zli82016** [comment]
Hello, @harshithpatte-g , is there a Github issue for this PR? If yes, can you please add the issue to the description section of this PR? Thanks.

File: mmv1/templates/terraform/examples/compute_interconnect_basic_test.tf.erb [hunk 1]
```diff
@@ -3,7 +3,7 @@ data "google_project" "project" {}
 resource "google_compute_interconnect" "<%= ctx[:primary_resource_id] %>" {
   name                 = "<%= ctx[:vars]['interconnect_name'] %>"
   customer_name        = "internal_customer" # Special customer only available for Google testing.
-  interconnect_type    = "IT_PRIVATE"        # Special type only available for Google testing.
+  interconnect_type    = "DEDICATED"        # Special type only available for Google testing.
```
Comment (zli82016) [inline mmv1/templates/terraform/examples/compute_interconnect_basic_test.tf.erb:6]: Is the comment `# Special type only available for Google testing.` still be correct for `DEDICATED `?

File: mmv1/templates/terraform/examples/interconnect_attachment_dedicated.tf.erb [hunk 2]
```diff
@@ -3,7 +3,7 @@ data "google_project" "project" {}
 resource "google_compute_interconnect" "foobar" {
   name                 = "<%= ctx[:vars]['interconnect_name'] %>"
   customer_name        = "internal_customer" # Special customer only available for Google testing.
-  interconnect_type    = "IT_PRIVATE"        # Special type only available for Google testing.
+  interconnect_type    = "DEDICATED"        # Special type only available for Google testing.
```
Comment (zli82016) [inline mmv1/templates/terraform/examples/interconnect_attachment_dedicated.tf.erb:6]: Is the comment `# Special type only available for Google testing.` still be correct for `DEDICATED `?

Comment (harshithpatte-g) [reply mmv1/templates/terraform/examples/interconnect_attachment_dedicated.tf.erb:6]: I will remove the comment. I don't think it is valid now.

**Comment by zli82016** [comment]
The test `TestAccComputeInterconnectAttachment_interconnectAttachmentDedicatedExample` failed with internal error. Can you please help to debug the error? Thanks.

```
{
 "kind": "compute#operation",
 "id": "8119474705099619496",
 "name": "operation-1721156166710-61d61e5eb762f-d7196e2e-086e85ea",
 "operationType": "insert",
 "targetLink": "https://www.googleapis.com/compute/beta/projects/ci-test-project-188019/global/interconnects/tf-test-interconenct-1lda9mdokaa",
 "targetId": "8318794671199059112",
 "status": "DONE",
 "user": "hashicorp-test-runner@ci-test-project-188019.iam.gserviceaccount.com",
 "progress": 100,
 "insertTime": "2024-07-16T11:56:07.073-07:00",
 "startTime": "2024-07-16T11:56:07.078-07:00",
 "endTime": "2024-07-16T11:56:10.107-07:00",
 "error": {
  "errors": [
   {
    "code": "INTERNAL_ERROR",
    "message": "Internal error. Please try again or contact Google Support. (Code: '-9184519272534638160')"
   }
  ]
 },
 "httpErrorStatusCode": 503,
 "httpErrorMessage": "SERVICE UNAVAILABLE",
 "selfLink": "https://www.googleapis.com/compute/beta/projects/ci-test-project-188019/global/operations/operation-1721156166710-61d61e5eb762f-d7196e2e-086e85ea"
}
```

File: mmv1/templates/terraform/constants/interconnect.go.erb [hunk 3]
```diff
@@ -0,0 +1,3 @@
+func InterconnectTypeDiffSuppress(k, old, new string, d *schema.ResourceData) bool {
+    return old == "DEDICATED" && new == "IT_PRIVATE"
```
Comment (zli82016) [inline mmv1/templates/terraform/constants/interconnect.go.erb:2]:
```suggestion
    return old == "IT_PRIVATE" && new == "DEDICATED"
```

**Review by zli82016 — CHANGES_REQUESTED** [review]

**Comment by harshithpatte-g** [comment]
> The test `TestAccComputeInterconnectAttachment_interconnectAttachmentDedicatedExample` failed with internal error. Can you please help to debug the error? Does this test pass locally? Thanks.
> 
> ```
> {
>  "kind": "compute#operation",
>  "id": "8119474705099619496",
>  "name": "operation-1721156166710-61d61e5eb762f-d7196e2e-086e85ea",
>  "operationType": "insert",
>  "targetLink": "https://www.googleapis.com/compute/beta/projects/ci-test-project-188019/global/interconnects/tf-test-interconenct-1lda9mdokaa",
>  "targetId": "8318794671199059112",
>  "status": "DONE",
>  "user": "hashicorp-test-runner@ci-test-project-188019.iam.gserviceaccount.com",
>  "progress": 100,
>  "insertTime": "2024-07-16T11:56:07.073-07:00",
>  "startTime": "2024-07-16T11:56:07.078-07:00",
>  "endTime": "2024-07-16T11:56:10.107-07:00",
>  "error": {
>   "errors": [
>    {
>     "code": "INTERNAL_ERROR",
>     "message": "Internal error. Please try again or contact Google Support. (Code: '-9184519272534638160')"
>    }
>   ]
>  },
>  "httpErrorStatusCode": 503,
>  "httpErrorMessage": "SERVICE UNAVAILABLE",
>  "selfLink": "https://www.googleapis.com/compute/beta/projects/ci-test-project-188019/global/operations/operation-1721156166710-61d61e5eb762f-d7196e2e-086e85ea"
> }
> ```

This is passing locally.
I had gotten this error when a project already has an interconnect.

File: mmv1/templates/terraform/constants/interconnect.go.erb (hunk 3 above)
Comment (harshithpatte-g) [reply mmv1/templates/terraform/constants/interconnect.go.erb:2]: I had initially tried with suggested code, it didn't work. 
OLD was DEDICATED and NEW was IT_PRIVATE, after the code changed to the current code, it worked.

Comment (zli82016) [reply mmv1/templates/terraform/constants/interconnect.go.erb:2]: The test `TestAccComputeInterconnect_computeInterconnectBasicTestExample` still failed with [permadiff](https://fff31dccb52217de7e62f72b6f6a648527ec25dd2f78d2187aab9fc-apidata.googleusercontent.com/download/storage/v1/b/ci-vcr-logs/o/beta%2Frefs%2Fheads%2Fauto-pr-11176%2Fartifacts%2Fd1a40a59-50f6-4ee8-bc1f-da0545968f3d%2Fbuild-log%2Frecording_build%2FTestAccComputeInterconnect_computeInterconnectBasicTestExample_recording_test.log?jk=AYBlUPB91QtWrT0tTlWgeCpeQfOvzKBxp5-UF822ho-OXl8H5P-1Eu-TFEMrAIj49UESp7HV59-kgWtOV0wW6Rt5X9IzNI6MvJo-2YxgG8AFG9dx5y4c2KLQrjIWe-ml8qXnd9xHD8us3j45804UzSt-sqJRsO8l3f4wTWwH7KfiyJSDgCWrKYSF7_MarS7CXV7YusudUtZPnair_3jjM8sZ1b7bqGhCjMkjG_M7HrE3w1TZRqEm-80eZDU8mzJQpPbBt40KRJfZRdDcA2QR8lG_IkHKOxA6K-ZI6gZ4oyVLuOTX5RHtlcNTAsPhvJfK8Ez7VsKjP4MBhs7iFdZuD34qq21iXYZjmZVgD-d57PXzYweIuLXTO3gJESaFcJHKHMm3xkSLkDCtSIVRrrJNC9-fzS2q7XuTsJo82FDHLW0zw5D-WXFDNjrhj5xGX-cGabPIBSGhiVuexYOtsuHLk7rs2rEsaRrlKaS2gQxFAXerFXb-vNJg21cU7yvLYfmFrNrarynvy-pAwaeT2GcL0Btw8-iMp-GXh3y8NWXPZbTp9qgq2EA4zM3rp5t7XVWNasDAcXqgRVmaH8bc2-2AGtgvZRzI7U3F1L-elT2HyFHQbW99puQO8J9VqXbdTjq4b_IsyVc3CtLMDDGU0MU6_02mxaVA5Ll_GadtNts0kay5UcrhVhiEjXqkpSIo_FTag5RfASXKC9eOuy4q8nsDuwuYhTzS8iZbQWX3TvpI5cujvzqPG4OwMcQjoAvtM751VD6qDfLNqaIAZUvZbOyWyHDJ9YGWsue9VugNItWBFdS9WDBsQIMWYJ4Ldw21JVpyEIIsN7o6yYY7Eu9LuFiNWzwENJzB0o9iMpNLhvKofmL2XQxX_imI3dX7G1Oj6bcAnpcPx69toE8KvgHRo0wZcLqGasOlWFC1_N5ym3Yn923ZDFE4H9jN7vaTwqCa1yLsXcn5T6cRNvKYm8S8X9q8Hce1GwbgenWTlujadgLavBiAh7HIMtzSQmnxa_bl4T8xzaEoP-PxF-gRhJlatQJml6NT9abgKnB3DXTKGAT0vBg62JGT7_V0IUlknpIrh_04_t08eoMHsN_Cv9OMiq4RYtVvcNF3Bo7YhKiIvo1EcSRsN3GCH1_XN582gNTVPnHrAyJIKxpadhHgm2-yksn5XK7o44gSnUzeHy25PsZL1mbJVjDvnhrwXyzufhimJ5CiOfjMGtdWlLuEomYBNJAmKF6t0lMMAmWLbAHu7uRgLwoT3HRK8zdQjf1KB2ytre3oj5_yE71ihC3m1kmhYNChbGeqY9xD2VKXM2qWL5EjXwtTtMSsKI_5CjUG0vc7Xm1avj9-CBgyi_SlUvfMovIOLU2uA7Iia9g07vyCv2CucRM&isca=1)

Comment (harshithpatte-g) [reply mmv1/templates/terraform/constants/interconnect.go.erb]: @zli82016 pushed the fix.

**Review by zli82016 — APPROVED** [review]
LGTM. Thanks.

**Pull request merged**